plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -61,16 +65,21 @@ const STEPS_TO_EMIT_TO_DLQ_ON_FAILURE: Array<StepType> = [
 
 export class EventPipelineRunner {
     hub: Hub
-    originalEvent: PluginEvent | ProcessedPluginEvent
+    originalEvent: PipelineEvent | ProcessedPluginEvent
 
-    constructor(hub: Hub, originalEvent: PluginEvent | ProcessedPluginEvent) {
+    constructor(hub: Hub, originalEvent: PipelineEvent | ProcessedPluginEvent) {
         this.hub = hub
         this.originalEvent = originalEvent
     }
 
-    async runEventPipeline(event: PluginEvent): Promise<EventPipelineResult> {
+    async runEventPipeline(event: PipelineEvent): Promise<EventPipelineResult> {
         this.hub.statsd?.increment('kafka_queue.event_pipeline.start', { pipeline: 'event' })
-        const result = await this.runPipeline('emitToBufferStep', event)
+        let result: EventPipelineResult
+        if (typeof event.team_id === 'number') {
```

Comment:
I would strongly suggest keeping business logic out of this class and within the individual steps if at all possible.

Business logic that's spread all over the place is tricky to follow.